packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -217,7 +218,42 @@ export class BlobManager {
                 JSON.stringify(Array.from(this.redirectTable.entries()))),
             );
         }
-        return { entries };
+
+        return this.summarizeEntries(entries);
+    }
+
+    private summarizeEntries(
+        entries: ITreeEntry[],
+    ): ISummarizeResult {
+        const builder = new SummaryTreeBuilder();
+        for (const entry of entries) {
+            switch (entry.type) {
+                case TreeEntry.Blob: {
```

Comment:
This is too generic function - there will never be TreeEntry.Blob for summaries produced by BlobManager.
I'd prefer this logic to be inlined into summarize() and leave only relevant parts.